### PR TITLE
[PWX-24563] Implemented Create() and Delete() APIs of storage interface for oracle cloud

### PR DESCRIPTION
**Pull request by vinayakshnd**
Signed-off-by: Vinayak Shinde <vinayakshnd@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: oracle/oracle.go [hunk 1]
```diff
@@ -366,3 +369,91 @@ func (o *oracleOps) Inspect(volumeIds []*string) ([]interface{}, error) {
...
+	createVolResp, err := o.storage.CreateVolume(context.Background(), createVolReq)
+	if err != nil {
+		return nil, err
+	}
+
+	err = o.waitVolumeStatus(*createVolResp.Id, core.VolumeLifecycleStateAvailable)
+	if err != nil {
+		return nil, o.rollbackCreate(*createVolResp.Id, err)
+	}
+	return o.refreshVol(createVolResp.Id)
```
Comment (nrevanna) [inline oracle/oracle.go:400]: refreshVol seems to be an extra unnecessary call. Make waitVolumeStatus return getVolResp and extract what you need there. That can be retuned. I am not sure why we have it for aws even.

File: oracle/oracle.go [hunk 2]
```diff
@@ -366,3 +367,90 @@ func (o *oracleOps) Inspect(volumeIds []*string) ([]interface{}, error) {
...
+			return nil, true, err
+		}
+		if getVolResp.Volume.LifecycleState == core.VolumeLifecycleStateAvailable {
+			return nil, false, nil
+		}
+
+		logrus.Debugf("volume [%s] is still in [%s] state", volID, getVolResp.Volume.LifecycleState)
+		return nil, true, fmt.Errorf("volume [%s] is still in [%s] state", volID, getVolResp.Volume.LifecycleState)
+	}
+	_, err := task.DoRetryWithTimeout(f, cloudops.ProviderOpsTimeout, cloudops.ProviderOpsRetryInterval)
```
[the remark this answers is not in the document]
Comment (vinayakshnd) [reply oracle/oracle.go:419]: From my experiance, Oracle takes around 6 to 8 Sec. for a volume to get created.

File: oracle/oracle.go [hunk 3]
```diff
@@ -366,3 +367,90 @@ func (o *oracleOps) Inspect(volumeIds []*string) ([]interface{}, error) {
...
+	}
+	return createErr
+}
+
+// Delete volumeID.
+func (o *oracleOps) Delete(volumeID string) error {
+	delVolReq := core.DeleteVolumeRequest{
+		VolumeId: &volumeID,
+	}
+	_, err := o.storage.DeleteVolume(context.Background(), delVolReq)
```
[the remark this answers is not in the document]
Comment (vinayakshnd) [reply oracle/oracle.go]: Done

File: oracle/oracle.go (hunk 1 above)
Comment (vinayakshnd) [reply oracle/oracle.go]: Done

File: oracle/oracle.go [hunk 4]
```diff
@@ -366,3 +367,90 @@ func (o *oracleOps) Inspect(volumeIds []*string) ([]interface{}, error) {
...
+			"Invalid volume template given", "")
+	}
+
+	createVolReq := core.CreateVolumeRequest{
+		CreateVolumeDetails: core.CreateVolumeDetails{
+			CompartmentId:      &o.compartmentID,
+			AvailabilityDomain: &o.availabilityDomain,
+			SizeInGBs:          vol.SizeInGBs,
+			VpusPerGB:          vol.VpusPerGB,
+			DisplayName:        vol.DisplayName,
```
[the remark this answers is not in the document]
Comment (vinayakshnd) [reply oracle/oracle.go:387]: Done

[the remark this answers is not in the document]
Comment (vinayakshnd) [reply oracle/oracle.go:387]: Done, whatever `labels` input arg is passed to `Create()` API, freeformTags will get created.

File: oracle/oracle.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrevanna) [reply oracle/oracle.go]: Got it. Thanks

**Review by nrevanna — APPROVED** [review]
Looks good to me code wise. Once you are done with CreateAndAttach we can test this. For now it's good to check in.

**Pull request merged**